src/Google.Api.Gax/Api/Gax/PathTemplate.cs

```diff
@@ -0,0 +1,440 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Represents a path template used for resource names which may be composed of multiple IDs.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Templates use a subset of the syntax of the API platform. See the protobuf of <see cref="HttpRule"/> for
+    /// details of the API platform. (The source for <c>HttpRule</c> is at
+    /// https://github.com/googleapis/googleapis/blob/master/google/api/http.proto)
+    /// </para>
+    /// <para>
+    /// This class performs no URL escaping or unescaping. It is designed for use within GRPC, where no
+    /// URL encoding is required.
+    /// </para>
+    /// </remarks>
+    public sealed class PathTemplate
+    {
+        /// <summary>
+        /// Just an array containing a single slash, to avoid constructing a new array every time we need
+        /// to split.
+        /// </summary>
+        private static readonly char[] SlashSplit = { '/' };
```

Comment:
s_slashSplit
If we follow https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md